receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -83,13 +97,48 @@ func (ghs *githubScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 		ghs.logger.Sugar().Debugf("using search query where query is: %v", ghs.cfg.SearchQuery)
 	}
 
-	repos, count, err := ghs.getRepos(ctx, genClient, sq)
```

Comment:
I'm thinking the top level should maintain this simplicity and the conditional logic should be under the hood (within the function)